pennylane/templates/layers/particle_conserving_u2.py

```diff
@@ -246,3 +255,67 @@ def shape(n_layers, n_wires):
                 f"The number of qubits must be greater than one; got 'n_wires' = {n_wires}"
             )
         return n_layers, 2 * n_wires - 1
+
+
+def _particle_conserving_u2_resources(num_wires: int, n_layers: int):
+    # number of pairs of even-indexed of wires
+    num_nm_wires = (
+        math.floor(num_wires / 2) if num_wires % 2 == 0 else math.ceil((num_wires - 1) / 2)
+    )
```

Comment:
I think this works the same? :) 
If `num_wires` is even, write it as `2k`. Then `num_wires//2=k`, which is equal to the current `floor(2k/2)=floor(k)`.
If `num_wires` is odd, write it as `2k+1`. Then `num_wires//2 = k` (`//` always rounds down to the next integer), which is equal to the current `ceil((2k+1-1)/2)=ceil(k)`.

```suggestion
    num_nm_wires = num_wires // 2
```